aiohttp/client_proto.py

```diff
@@ -240,6 +240,14 @@ def _reschedule_timeout(self) -> None:
     def start_timeout(self) -> None:
         self._reschedule_timeout()
 
+    @property
```

Comment:
Ideally this would get set earlier so it doesn't need to be exposed as a property and setter.

Reply:
It can't be. That's what differs between HTTP connection and WS connection and that's why the bug exists.

When we upgrade from HTTP to WS the original HTTP request with the upgrade header needs to be done under the socket timeouts of HTTP connections. Once upgrade is established we need to *switch* connection timeout parameters to the WS mode. The crux is the upgrade.

Reply:
Ideally this would only be exposed for websocket connections as we need to take a look at all the places this is used and make sure it works as expected for everything since this exposes a new public setter.

Reply:
Any suggestions? I unfortunately don't have time to dedicate to solving all the issues. 
We can make the property `_ws_read_timeout` to indicate specialization.
As-is this is absolutely harmless IMO.

Could I perhaps suggest this as an interim solution (since it works) and a followup ticket will "re-architect" this issue unless it's something obvious/trivial?

Reply:
I looked at all the usage, and changing it after construction doesn't seem like its a problem since we already allow changing it via `set_response_params` so I think its OK.  

The downside is that we have another method to maintain but at least its simple.

Reply:
If we want to make the attribute public, shouldn't we just rename it?